src/main/java/com/excelsiorjet/api/tasks/JetProject.java

```diff
@@ -643,13 +641,116 @@ public JetHome validate(boolean validateForBuild) throws JetTaskFailureException
             execProfilesName = projectName;
         }
         if (validateForBuild) {
-            validateForBuild(jetHomeObj);
+            validateForBuild(excelsiorJet);
         }
 
-        return jetHomeObj;
+        processDependencies();
     }
 
-    private void validateForBuild(JetHome jetHomeObj) throws JetTaskFailureException {
+    public void processDependencies() throws JetTaskFailureException {
+        for (DependencySettings dependency : dependencies) {
+            if (!dependency.isValid()) {
+                throw new JetTaskFailureException(s("JetApi.InvalidDependencySetting", dependency.idStr()));
+            }
+        }
+        List<DependencySettings> dependenciesSettings = dependencies.stream()
+                .filter(DependencySettings::isSettings)
```

Comment:
We should simply check all erroneous cases and issue error via validating. After validating the predicates become mutual exclusive. Keep isExternal